### PR TITLE
[WIP] INT96 changes for Parquet writer [skip ci]

### PR DESCRIPTION
**Pull request by razajafri**
This PR is dependent upon #5096. 

Added changes to the following files 
ParquetWriterOptions.java 
TableJni.cpp
TableTest.java
Table.java

CMakeLists.txt (unrelated to INT96 change but needed to build-cudf-java inside the compose container). If this is a concern I can gladly split this into its own PR

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: java/src/main/java/ai/rapids/cudf/ParquetWriterOptions.java [hunk 1]
```diff
@@ -63,9 +72,19 @@ public static Builder builder() {
...
   }
 
   public StatisticsFrequency getStatisticsFrequency() {
     return statsGranularity;
   }
+
+  /**
+   * Returns true if the writer is expected to write timestamps in INT96
+   */
+  public boolean isInt96() {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ParquetWriterOptions.java:85]: nit, I would prefer this to be a slightly more descriptive function name. It does not mention timestamp or anything and it is a public method.

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 2]
```diff
@@ -257,6 +257,7 @@ private static native long writeParquetBufferBegin(String[] columnNames,
                                                      String[] metadataValues,
                                                      int compression,
                                                      int statsFreq,
+                                                     boolean isInt96,
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Table.java:260]: Please update the javadocs

File: java/src/main/native/src/TableJni.cpp [hunk 3]
```diff
@@ -856,11 +856,12 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_readParquet(
 JNIEXPORT long JNICALL Java_ai_rapids_cudf_Table_writeParquetBufferBegin(
     JNIEnv *env, jclass, jobjectArray j_col_names, jbooleanArray j_col_nullability,
     jobjectArray j_metadata_keys, jobjectArray j_metadata_values, jint j_compression,
-    jint j_stats_freq, jobject consumer) {
+    jboolean j_isInt96, jint j_stats_freq, jobject consumer) {
   JNI_NULL_CHECK(env, j_col_names, "null columns", 0);
   JNI_NULL_CHECK(env, j_col_nullability, "null nullability", 0);
   JNI_NULL_CHECK(env, j_metadata_keys, "null metadata keys", 0);
   JNI_NULL_CHECK(env, j_metadata_values, "null metadata values", 0);
+  JNI_NULL_CHECK(env, j_isInt96, "null isInt96 values", 0);
```
Comment (revans2) [inline java/src/main/native/src/TableJni.cpp:864]: a jboolean cannot be null it is a basic type, like boolean.  I think this assertion will fail if j_isInt96 is false.

File: java/src/main/native/src/TableJni.cpp [hunk 4]
```diff
@@ -887,6 +888,7 @@ JNIEXPORT long JNICALL Java_ai_rapids_cudf_Table_writeParquetBufferBegin(
         .nullable_metadata(&metadata)
         .compression(static_cast<compression_type>(j_compression))
         .stats_level(static_cast<statistics_freq>(j_stats_freq))
+        //.int96_timestamps(true)
```
Comment (revans2) [inline java/src/main/native/src/TableJni.cpp:891]: please commencement this and set it to the correct value.  the dependency has been merged in.

**Pull request closed without merging**